x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/data_input_flyout/steps/rules/sub_steps/migration_name/migration_name_input.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useCallback, useMemo, useEffect, useRef } from 'react';
+import { EuiFieldText, EuiFlexGroup, EuiFlexItem, EuiFormRow, EuiForm } from '@elastic/eui';
+import * as i18n from './translations';
+
+export interface MigrationNameInputProps {
+  migrationName: string;
+  setMigrationName: (migrationName: string) => void;
+  subStep: number;
+  defaultMigrationName: string;
+}
+
+export const MigrationNameInput = React.memo<MigrationNameInputProps>(
+  ({ migrationName, setMigrationName, subStep, defaultMigrationName }) => {
+    const [name, setName] = useState<string>(defaultMigrationName);
```

Comment:
This component looks great, let's clean unnecessary props:

```suggestion
export interface MigrationNameInputProps {
  migrationName: string;
  setMigrationName: (migrationName: string) => void;
}

export const MigrationNameInput = React.memo<MigrationNameInputProps>(
  ({ migrationName, setMigrationName }) => {
    const [name, setName] = useState<string>(migrationName);
```